allo/backend/hls.py

```diff
@@ -168,6 +170,39 @@ def __init__(
             allo_d.register_dialect(ctx)
             self.module = Module.parse(str(mod), ctx)
             self.func = find_func_in_module(self.module, top_func_name)
+
+            if platform == "ihls":
+                assert mode in {
+                    "fpga_emulator",
+                    "source_file_only",
+                    None,
+                }, "Invalid mode"
+
+                if project is not None:
+                    filename = f"{project}"
```

Comment:
filename is different from `project`. For example, users can specify the `project` as a nested folder `test/out.prj`.